later/lib/database/repository/record_repository.dart

```diff
@@ -1,37 +1,71 @@
-import 'dart:async';
-
 import 'package:injectable/injectable.dart';
-import 'package:isar/isar.dart';
-import 'package:later/database/models/record_model.dart';
+import 'package:later/database/datasources/records_data_source.dart';
+import 'package:later/domain/models/record_model.dart';
+import 'package:later/domain/repository/record_repository.dart';
 
 @lazySingleton
-class RecordRepository {
-  late final Isar _isar;
+class RecordRepositoryImpl implements RecordsRepository {
+  RecordRepositoryImpl({
+    required this.recordsDataSurce,
+  });
 
-  // We must preinitialize it first
-  @FactoryMethod(preResolve: true)
-  static Future<RecordRepository> init() async {
-    final repo = RecordRepository()
-      .._isar = await Isar.open([RecordModelSchema]);
+  final RecordsDataSource recordsDataSurce;
 
-    return repo;
+  @override
+  Future<void> create({
```

Comment:
For `create` method, prefer returning a model. E.g. this should create a `RecordModel` as a result. For this:
- Make `RecordDataSource` to return a DB model as well. 
- Map it here to a `RecordModel`